### PR TITLE
neo.units module (Issue278)

### PR DESCRIPTION
**Pull request by joffreygonin**
Create `neo.units` module #278

### CODE REVIEW

**Comment by apdavison** [comment]
@jgonin Please can you resolve the conflicts by rebasing your changes onto the latest master (see https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request for how to do this, replacing "edx/edx-platform" with "NeuralEnsemble/python-neo")

File: neo/core/event_BACKUP_4347.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
+# -*- coding: utf-8 -*-
```
Comment (apdavison) [inline neo/core/event_BACKUP_4347.py]: This files should not be here - there are other similar files generated (I think) by Git merging or rebase operations, which have been added to the commit and need to be removed.

File: doc/source/core.rst [hunk 2]
```diff
@@ -66,7 +66,7 @@ cut across the simple container hierarchy.
 NumPy compatibility
 ===================
 
-Neo data objects inherit from :py:class:`Quantity`, which in turn inherits from NumPy
+Neo data objects inherit from :py:class:`Quantity` (by using units module), which in turn inherits from NumPy
```
Comment (apdavison) [inline doc/source/core.rst]: revert - we should not import Quantity via the `units` module

File: doc/source/images/generate_diagram.py [hunk 3]
```diff
@@ -173,7 +174,7 @@ def generate_diagram(filename, rect_pos, rect_width, figsize):
             else:
                 t1 = attrname
 
-            if attrtype == pq.Quantity:
+            if attrtype == un.Quantity:
```
Comment (apdavison) [inline doc/source/images/generate_diagram.py]: Use the `units` module only for importing units, not for the `quantities.Quantity` class. This should still be `pq.Quantity`

File: neo/core/analogsignal.py [hunk 4]
```diff
@@ -64,7 +65,7 @@ def _new_AnalogSignalArray(cls, signal, units=None, dtype=None, copy=True,
                **annotations)
 
 
-class AnalogSignal(BaseNeo, pq.Quantity):
+class AnalogSignal(BaseNeo, un.Quantity):
```
Comment (apdavison) [inline neo/core/analogsignal.py]: should stay as `pq.Quantity`

**Comment by samuelgarcia** [comment]
@joffreygonin: rawio is now the master.
As discussed when I was in Gif, I propose to close this PR because it is not compatible with the new master. So you an start working on this in a new branch.

**Pull request closed without merging**